src/client/sandbox/node/window.js

```diff
@@ -564,6 +569,41 @@ export default class WindowSandbox extends SandboxBase {
             });
         }
 
+        const windowSandbox = this;
+
+        overrideDescriptor(window.HTMLInputElement.prototype, 'files', function () {
+            if (this.type.toLowerCase() === 'file')
+                return UploadSandbox.getFiles(this);
+
+            return nativeMethods.inputFilesGetter.call(this);
+        });
+
+        overrideDescriptor(window.HTMLInputElement.prototype, 'value', function () {
+            if (this.type.toLowerCase() === 'file')
+                return UploadSandbox.getUploadElementValue(this);
+
+            return nativeMethods.inputValueGetter.call(this);
+        }, function (value) {
+            if (this.type.toLowerCase() === 'file')
+                return windowSandbox.uploadSandbox.setUploadElementValue(this, value);
+
+            nativeMethods.inputValueSetter.call(this, value);
+
+            if (isTextEditableElementAndEditingAllowed(this) && !isShadowUIElement(this))
```

Comment:
Replace with `if (!isShadowUIElement(this) && isTextEditableElementAndEditingAllowed(this))` because `!isShadowUIElement(this)` is faster.
Here and below.